crates/bevy_ecs/src/query/state.rs

```diff
@@ -78,6 +80,7 @@ where
         }
     }
 
+    /// Validates a query for the given [`World`] and updates all archetypes.
```

Comment:
This comment just feels like it's restating the name of the function.
Perhaps, the comment could better describe what the function actually does?
What does it mean to `update an archetype` and what does it mean to `validate the world`?